quixstreams/dataframe/windows/time_based.py

```diff
@@ -231,3 +247,240 @@ def wrapper(
         return func(value, key, timestamp, headers, state)
 
     return wrapper
+
+
+class SlidingWindow(FixedTimeWindow):
```

Comment:
Let's move it to a separate module since it's quite big.
We can convert `time_based.py` to a module for that purpose too